File: 383-ransom-note/383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        if(ransomNote.length() > magazine.length()){
            return false;
        }
        int match = 0;
        Map<Character, Integer> mapRanso = new HashMap<>();
        Map<Character, Integer> mapMagazine = new HashMap<>();
        
        for(char ch : ransomNote.toCharArray()){
            mapRanso.put(ch, mapRanso.getOrDefault(ch, 0)+1);
        }
        
        for(char ch : magazine.toCharArray()){
            mapMagazine.put(ch, mapMagazine.getOrDefault(ch, 0)+1);
        }
        for(var entry : mapRanso.entrySet()){
            if(mapMagazine.containsKey(entry.getKey())){
                if(mapMagazine.get(entry.getKey()) < entry.getValue()){
                    return false;
                }else{
                    match+=entry.getValue();
                }
            }            
        }
        return match == ransomNote.length();
    }

}

File: 383-ransom-note/383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        if(len(ransomNote) > len(magazine)):
            return False
        
        ranCont, magCont = [0]*26, [0]*26
        
        for i in range(len(ransomNote)):
            ranCont[ord(ransomNote[i]) - ord('a')]+=1
        
        for i in range(len(magazine)):
            magCont[ord(magazine[i]) - ord('a')]+=1
        
        
        match = 0
        for i in range(26):
            if ranCont[i] <= magCont[i]:
                match +=ranCont[i]
                
        return match == len(ransomNote)